packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -709,13 +711,32 @@ export default function makeKernelKeeper(
     return owner;
   }
 
-  function orphanKernelObject(kref, oldVat) {
+  function retireKernelObjects(koids) {
+    Array.isArray(koids) || Fail`retireExports given non-Array ${koids}`;
+    const newActions = [];
+    for (const koid of koids) {
+      const importers = getImporters(koid);
+      for (const vatID of importers) {
+        newActions.push(`${vatID} retireImport ${koid}`);
+      }
+      deleteKernelObject(koid);
+    }
+    addGCActions(newActions);
+  }
+
+  function abandonKernelObject(kref, oldVat) {
```

Comment:
This function removes a kref from its owning vat's c-list and clears its owner, which seems better described by the old name of "orphanKernelObject" than by the new one of "abandonKernelObject".
```suggestion
  function orphanKernelObject(kref, oldVat) {
```
But if it's important to use a new name to avoid confusion of old references, then might I suggest
```suggestion
  function disownKernelObject(kref, oldVat) {
```

Reply:
yeah, `orphan` sounds good now